cmd/utils.go

```diff
@@ -178,37 +176,46 @@ type prefixSSEPair struct {
 
 // parse list of comma separated alias/prefix=sse key values entered on command line and
 // construct a map of alias to prefix and sse pairs.
-func parseEncryptionKeys(ssekeys string) (encMap map[string][]prefixSSEPair, err *probe.Error) {
+func parseEncryptionKeys(sseKeys string) (encMap map[string][]prefixSSEPair, err *probe.Error) {
 	encMap = make(map[string][]prefixSSEPair)
-	if ssekeys == "" {
+	if sseKeys == "" {
 		return
 	}
-	fields := strings.Split(ssekeys, ",")
-	r := regexp.MustCompile("([^=\\s]+)=([^\\s]+)")
-	for _, field := range fields {
-		field = strings.TrimSpace(field)
-		matches := r.FindStringSubmatch(field)
-		if len(matches) != 3 {
-			return nil, probe.NewError(errors.New("sse-c prefix should be of the form prefix=key"))
+	prefix := ""
+	ssekey := ""
+	index := 0 // start index of prefix
+	vs := 0    // start index of sse-c key
+	sseKeyLen := 32
+	delim := 1
+	l := len(sseKeys)
```

Comment:
suggestion: can we use a different single letter variable? `l` can be easily mistaken for `1`.

Reply:
done.